### PR TITLE
Create Dns Xact Handler

### LINKED ISSUE #341: DNS Handler - Create DnsHandler V2
The DNS handler should still be able to provide `xact` metrics when an `rcode` filter is applied.

Currently, if a filter is applied such as `"only_rcode": 3` (only NXDOMAIN), the `xact` metrics are empty (all zeroes).

### PR DESCRIPTION
**Pull request by leoparente**
- Create Common Visor DNS Headers
- Remove all transaction logic from DNS Handler
- Create Own DNS Xact Handler
- Some filters should only be performed if the packet is response (so transaction can still be tracked)

### CODE REVIEW

**Comment by leoparente** [comment]
**Improvements questions**: on DNS Xact Will Filter we should delete transaction that will be filtered out? I think So..

File: src/handlers/dns_xact/DnsXactStreamHandler.h [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+        Counter xacts_in;
+        Counter xacts_out;
+        Counter xacts_timed_out;
+        Counter filtered;
+        counters()
+            : xacts_total("dns", {"xact", "counts", "total"}, "Total DNS transactions (query/reply pairs)")
+            , xacts_in("dns", {"xact", "in", "total"}, "Total ingress DNS transactions (host is server)")
+            , xacts_out("dns", {"xact", "out", "total"}, "Total egress DNS transactions (host is client)")
+            , xacts_timed_out("dns", {"xact", "counts", "timed_out"}, "Total number of DNS transactions that timed out")
+            , filtered("dns", {"wire_packets", "filtered"}, "Total DNS wire packets seen that did not match the configured filter(s) (if any)")
```
Comment (rboucher-me) [inline src/handlers/dns_xact/DnsXactStreamHandler.h]: For consistency, I'm thinking we should name this metric something like `dns_xact_packets_filtered` (or `dns_xact_wire_packets_filtered`). It's not ideal but at least it's consistent.

File: src/handlers/dns_xact/README.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# DNS Stream Handler
```
Comment (rboucher-me) [inline src/handlers/dns_xact/README.md]: Should this be the "DNS Transaction (xact) Stream Handler"?

Comment (leoparente) [reply src/handlers/dns_xact/README.md]: Yes, it should

Comment (leoparente) [reply src/handlers/dns_xact/README.md]: done

File: src/handlers/dns_xact/DnsXactStreamHandler.h (hunk 1 above)
Comment (leoparente) [reply src/handlers/dns_xact/DnsXactStreamHandler.h]: done as dns_xact_packets_filtered

**Comment by weyrick** [comment]
This is in Draft, is it ready for review?

**Comment by leoparente** [comment]
> This is in Draft, is it ready for review?

No, It is not. Things changed a lot on DnsHandler, rebase this will be difficult. I will start over.

**Pull request closed without merging**